<launch>

  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="LOGGER_DEBUG" default="$(optenv LOGGER_DEBUG false)" />

  <env name="ROSCONSOLE_CONFIG_FILE" if="$(eval arg('LOGGER_DEBUG'))" value="$(find octomap_tools)/config/debug_verbosity.yaml" />

    <!-- other args -->
  <arg name="debug" default="false" />
  <arg name="standalone" default="true" />
  <arg name="custom_config" default="" />

  <arg name="map_path" default="$(env HOME)/maps" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <arg name="nodelet_manager_name" default="" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg nodelet_manager_name)" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="octomap_editor" args="$(arg nodelet) octomap_tools/OctomapEditor $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <rosparam file="$(find octomap_tools)/config/octomap_editor.yaml" command="load" />

      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />

      <param name="map_path" type="string" value="$(arg map_path)" />

      <!-- topics out -->
      <remap from="~octomap_out" to="~octomap" />

    </node>

  </group>

  <include file="$(find octomap_tools)/launch/octomap_rviz_visualizer.launch">

    <arg name="standalone" value="$(arg standalone)" />
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)" />
    <arg name="custom_config" value="" />
    <arg name="debug" value="false" />
    <arg name="node_name" value="octomap_vis" />
    <arg name="octomap_topic" value="octomap_editor/octomap" />

  </include>

</launch>
